src/app/services/wallet.service.ts

```diff
@@ -314,32 +315,43 @@ export class WalletService {
   }
 
   loadBalances() {
+    if (this.updatingBalance) {
+      return;
+    }
+
+    this.updatingBalance = true;
+
     this.addresses.first().subscribe((addresses: Address[]) => {
-      this.retrieveAddressesBalance(addresses)
-        .subscribe((balance: Balance) => {
-          this.wallets.first().subscribe(wallets => {
-            if (balance.addresses) {
-              wallets.map((wallet: Wallet) => {
-                wallet.addresses.map((address: Address) => {
-                  if (balance.addresses[address.address]) {
-                    address.balance = balance.addresses[address.address].confirmed.coins / this.coinsMultiplier;
-                    address.hours = balance.addresses[address.address].confirmed.hours;
-                  }
-                });
-
-                wallet.balance = wallet.addresses.map(address => address.balance >= 0 ? address.balance : 0).reduce((a , b) => a + b, 0);
-                wallet.hours = wallet.addresses.map(address => address.hours >= 0 ? address.hours : 0).reduce((a , b) => a + b, 0);
-              });
-            }
-
-            this.calculateTotalBalance(wallets);
-            const hasPendingTxs = this.refreshPendingTransactions(balance);
-            this.resetBalancesUpdateTime(hasPendingTxs);
-          });
+      this.retrieveAddressesBalance(addresses).subscribe((balance: Balance) => {
+          this.wallets.first().subscribe(
+            wallets => this.calculateBalance(wallets, balance),
+            () => this.updatingBalance = false,
+            () => this.updatingBalance = false
+          );
         });
```

Comment:
Fixed:  `this.updatingBalance = false` is moved to the `complete` event of `this.retrieveAddressesBalance`.
Also `this.updatingBalance = false` is added in `complete` event due to `complete` event is not firing in case of any error.